botocore/utils.py

```diff
@@ -1491,6 +1491,9 @@ def func_with_weakref(weakref_to_self, *args, **kwargs):
 
         @functools.wraps(func)
         def inner(self, *args, **kwargs):
+            for kwarg_key, kwarg_value in kwargs.items():
+                if isinstance(kwarg_value, list):
+                    kwargs[kwarg_key] = tuple(kwarg_value)
```

Comment:
Have we manually tested that we get a cache hit when using operationContextParams?

Reply:
Yes, we manually tested this and we got a cache hit.

Reply:
Yes, we ran a test with some hypothetical models and ensured that the endpoint was not resolved twice.  This code is necessary to run before we check the cache, so these few lines do get called multiple times.